### PR TITLE
Make the mirador config more easily alterable, Drupal-side.

### PR DESCRIPTION
**Pull request by adam-vessey**
# What does this Pull Request do?

Moves building up the "object" of Mirador viewer config PHP-side, so we can alter/adjust/augment it more easily, minimizing the size of the actual Javascript "integration" piece required.

* **Related GitHub Issue**: (link)

* **Other Relevant Links**: (Google Groups discussion, related pull requests,
 Release pull requests, etc.)

# What's new?
A in-depth description of the changes made by this PR. Technical details and
 possible side effects.

* Changes x feature to such that y
* Added x
* Removed y
* Does this change add any new dependencies? 
* Does this change require any other modifications to be made to the repository
 (i.e. Regeneration activity, etc.)? 
* Could this change impact execution of existing code?

# How should this be tested?

A description of what steps someone could take to:
* Reproduce the problem you are fixing (if applicable)
* Test that the Pull Request does what is intended.
* Please be as detailed as possible.
* Good testing instructions help get your PR completed faster.

# Documentation Status

* Does this change existing behaviour that's currently documented?
* Does this change require new pages or sections of documentation?
* Who does this need to be documented for?
* Associated documentation pull request(s): ___  or documentation issue ___

# Additional Notes:
Any additional information that you think would be helpful when reviewing this
 PR.

# Interested parties
Tag (@ mention) interested parties or, if unsure, @Islandora/committers

### CODE REVIEW

File: js/mirador_viewer.js [hunk 1]
```diff
@@ -1,57 +1,30 @@
...
      * Initialize the Mirador Viewer.
      */
     Drupal.behaviors.Mirador = {
         attach: function (context, settings) {
-            base = settings.mirador_view_id;
-            init(context,settings);
+            Object.entries(settings.mirador.viewers).forEach(entry => {
+              const [base, values] = entry;
+              once('mirador-viewer', base, context).forEach(() =>
+                Mirador.viewer(values, window.miradorPlugins || {})
```
Comment (adam-vessey) [inline js/mirador_viewer.js:18]: The `window.miradorPlugins` thing here seems like we could have saner expectations? Like, have the integration itself include it itself, so there's not this arbitrary JS extension point?

Comment (adam-vessey) [reply js/mirador_viewer.js:18]: With something like https://github.com/roblib/mirador-integration-islandora/pull/1, could possibly avoid having to touch `window.miradorPlugins` here? Could just become:

```suggestion
                Mirador.viewer(values)
```

**Comment by jordandukart** [comment]
This looks fine here to me but given the in-flight work being undertook in the community I am hesitant to merge this prior to having more sets of eyes on this, primarily in regards to the `roblib` integration pull and the changing of the `attached` structure (albeit it's backwards compatability).

**Review by alxp — APPROVED** [review]

**Comment by alxp** [comment]
Just noticed that this was no longer marked as Draft.

I tested it and the plugins and everything still works as expected so this looks good.

**Pull request merged**